SDL/Quintuplet.cu

```diff
@@ -428,6 +428,117 @@ __device__ bool SDL::runQuintupletDefaultAlgo(struct SDL::modules& modulesInGPU,
+        sqrtf(x4*x4 + y4*y4),                                         // outer (hit 4) t3_2_r
+        float(modulesInGPU.layers[lowerModuleIndex4] + 6*is_endcap4), // outer (hit 4) t3_2_layer
+        mdsInGPU.anchorEta[fifthMDIndex],                             // outer (hit 5) t3_4_eta
+        mdsInGPU.anchorPhi[fifthMDIndex],                             // outer (hit 5) t3_4_phi
+        mdsInGPU.anchorZ[fifthMDIndex],                               // outer (hit 5) t3_4_z
+        sqrtf(x5*x5 + y5*y5),                                         // outer (hit 5) t3_4_r
+        float(modulesInGPU.layers[lowerModuleIndex5] + 6*is_endcap5), // outer (hit 5) t3_4_layer
+        log10((innerRadius+outerRadius)*3.8f*1.602f/(2*100*5.39f)),   // t5_pt
+        mdsInGPU.anchorEta[segmentsInGPU.mdIndices[2*tripletsInGPU.segmentIndices[2*innerTripletIndex+layer2_adjustment]]], // t5_eta
+        mdsInGPU.anchorPhi[segmentsInGPU.mdIndices[2*tripletsInGPU.segmentIndices[2*innerTripletIndex+layer2_adjustment]]], // t5_phi
+        log10(innerRadius),                                           // t5_innerRadius
+        log10(bridgeRadius),                                          // t5_bridgeRadius
+        log10(outerRadius)                                            // t5_outerRadius"
+    };
+
+    // (0): Linear(in_features=38, out_features=32, bias=True) => x = x*W_T + b
+    float x_0[32];
+    for (unsigned int col = 0; col < 32; ++col)
+    {
+        x_0[col] = 0;
+        for (unsigned int inner = 0; inner < 38; ++inner)
+        {
+            x_0[col] += x[inner]*T5DNN::wgtT_0[inner][col];
+        }
+        x_0[col] += T5DNN::bias_0[col];
+    }
+    
+    // (1): ReLU()
+    float x_1[32];
+    for (unsigned int col = 0; col < 32; ++col)
+    {
+        x_1[col] = (x_0[col] > 0.f) ? x_0[col] : 0.f;
+    }
+    
+    // (2): Linear(in_features=32, out_features=32, bias=True) => x = x*W_T + b
+    float x_2[32];
+    for (unsigned int col = 0; col < 32; ++col)
+    {
+        x_2[col] = 0;
+        for (unsigned int inner = 0; inner < 32; ++inner)
+        {
+            x_2[col] += x_1[inner]*T5DNN::wgtT_2[inner][col];
+        }
+        x_2[col] += T5DNN::bias_2[col];
+    }
+    
+    // (3): ReLU()
+    float x_3[32];
+    for (unsigned int col = 0; col < 32; ++col)
+    {
+        x_3[col] = (x_2[col] > 0.f) ? x_2[col] : 0.f;
+    }
+    
+    // (4): Linear(in_features=32, out_features=1, bias=True) => x = x*W_T + b
+    float x_4[1];
+    for (unsigned int col = 0; col < 1; ++col)
+    {
+        x_4[col] = 0;
+        for (unsigned int inner = 0; inner < 32; ++inner)
+        {
+            x_4[col] += x_3[inner]*T5DNN::wgtT_4[inner][col];
+        }
+        x_4[col] += T5DNN::bias_4[col];
+    }
+    
+    // (5): Sigmoid()
+    float x_5[1];
+    for (unsigned int col = 0; col < 1; ++col)
+    {
+        x_5[col] = exp(x_4[col])/(exp(x_4[col]) + 1);
```

Comment:
I was wondering whether 1/(1+exp(-x)) or exp(x)/(1+exp(x)) would be faster for the sigmoid. Maybe it's worth checking.
While searching about it, I also found this post:
https://stackoverflow.com/questions/10732027/fast-sigmoid-algorithm
Probably not something for this PR but could be tried?

Reply:
Interesting, though I'm not sure if we are sensitive to any speedup this might give, i.e. the for loops take so long that a small speedup in changing the sigmoid around is not noticeable. I can try at some point anyway, though, and the suggestion is interesting--thanks!